src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,82 @@
+            StringBuilder stringBuilder = new StringBuilder();
+            String value = bufferedReader.readLine();
+            while (value != null) {
+                stringBuilder.append(value).append("\n");
+                value = bufferedReader.readLine();
+            }
+            String stringBuilderToString = stringBuilder.toString();
+            String[] stringBuilderToStringArray = stringBuilderToString.split("\n");
+            for (int i = 0; i < stringBuilderToStringArray.length; i++) {
+                if (stringBuilderToStringArray[i].charAt(0) == 's') {
+                    countS++;
+                    if (countS == 1) {
+                        int index = stringBuilderToStringArray[i].indexOf(",");
+                        RESULT[0] = stringBuilderToStringArray[i].substring(0, index + 1);
+                    }
+                }
+                if (stringBuilderToStringArray[i].charAt(0) == 'b') {
+                    countB++;
+                    if (countB == 1) {
+                        int index = stringBuilderToStringArray[i].indexOf(",");
+                        RESULT[1] = stringBuilderToStringArray[i].substring(0, index + 1);
+                    }
+                }
+            }
+            for (int i = 0; i < stringBuilderToStringArray.length; i++) {
+                if (stringBuilderToStringArray[i].charAt(0) == 's') {
+                    int indexOf = stringBuilderToStringArray[i].indexOf(",");
+                    stringBuilderToStringArray[i] =
+                            stringBuilderToStringArray[i].substring(indexOf + 1);
+                    sumS = sumS + Integer.parseInt(stringBuilderToStringArray[i]);
 
+                }
+                if (stringBuilderToStringArray[i].charAt(0) == 'b') {
+                    int indexOf = stringBuilderToStringArray[i].indexOf(",");
+                    stringBuilderToStringArray[i] =
+                            stringBuilderToStringArray[i].substring(indexOf + 1);
+                    sumB = sumB + Integer.parseInt(stringBuilderToStringArray[i]);
+                }
+            }
+            difference = sumS - sumB;
+            RESULT[0] = RESULT[0] + sumS;
+            RESULT[1] = RESULT[1] + sumB;
+            RESULT[2] = "result," + difference;
+            File toFile = new File(toFileName);
+            for (String res : RESULT) {
+                try {
+                    BufferedWriter bufferedWriter =
+                            new BufferedWriter(new FileWriter(toFile, true));
+                    bufferedWriter.write(res);
+                    bufferedWriter.write(System.lineSeparator());
+                    bufferedWriter.close();
+                } catch (IOException e) {
+                    throw new RuntimeException("Can`t write data file", e);
+                }
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can`t read file", e);
+        }
     }
 }
```

Comment:
try to separate this method to smaller ones, for example one for read from file, second for generating report, third for write to file 
also, use try with resources if it's applicable 
Additionally , remember that you can make constants instead of writing "," every time etc
try to avoid magic numbers even if there are array indexes